app/helpers/__tests__/Intersect.spec.js

```diff
@@ -1,19 +1,569 @@
-import { isLocationsNearby } from '../Intersect';
+import dayjs from 'dayjs';
+import duration from 'dayjs/plugin/duration';
+import MockDate from 'mockdate';
 
-describe('isLocationsNearby', () => {
+import {
+  CONCERN_TIME_WINDOW_MINUTES,
+  DEFAULT_EXPOSURE_PERIOD_MINUTES,
+} from '../../constants/history';
+import {
+  areLocationsNearby,
+  getEmptyLocationBins,
+  intersectSetIntoBins,
+  normalizeAndSortLocations,
+} from '../Intersect';
 
-  // simple tests for the isLocationNearby function
+const TEST_MOMENT_MS = dayjs('2020-04-17T16:00:00.000Z').valueOf();
 
-  it('north and south poles not nearby',() => {
-      expect(isLocationsNearby(90,0,-90,0)).toBe(false);
+/**
+ * locations used in testing.  Set up as a single object to help simplify test setup.
+ *      Kansas City, MO, USA
+ *      Hammerfest, Norway
+ *      Hobart, Tasmania, Australia
+ *      Munich, Germany
+ *      La Concordia, Costa Rica
+ */
+const TEST_LOCATIONS = {
+  kansascity: {
+    base: {
+      lat: 39.09772,
+      lon: -94.582959,
+    },
+    concern: {
+      lat: 39.097769,
+      lon: -94.582937,
+    },
+    no_concern: {
+      lat: 39.1,
+      lon: -94.6,
+    },
+  },
+  hammerfest: {
+    base: {
+      lat: 70.663017,
+      lon: 23.682105,
+    },
+    concern: {
+      lat: 70.66302,
+      lon: 23.6822,
+    },
+    no_concern: {
+      lat: 70.662482,
+      lon: 23.68115,
+    },
+  },
+  hobart: {
+    base: {
+      lat: -42.8821,
+      lon: 147.3272,
+    },
+    concern: {
+      lat: -42.88215,
+      lon: 147.32721,
+    },
+    no_concern: {
+      lat: -42.883,
+      lon: 147.328,
+    },
+  },
+  munich: {
+    base: {
+      lat: 48.1351,
+      lon: 11.582,
+    },
+    concern: {
+      lat: 48.13515,
+      lon: 11.58201,
+    },
+    no_concern: {
+      lat: 48.136,
+      lon: 11.584,
+    },
+  },
+  laconcordia: {
+    base: {
+      lat: 0.01,
+      lon: -79.3918,
+    },
+    concern: {
+      lat: 0.01005,
+      lon: -79.39185,
+    },
+    no_concern: {
+      lat: 0.01015,
+      lon: -79.3912,
+    },
+  },
+};
+
+/**
+ *   Testing the intersect function.  Multiple cases covered.
+ */
+describe('intersectSetIntoBins', () => {
+  dayjs.extend(duration);
+
+  beforeEach(() => {
+    MockDate.set(TEST_MOMENT_MS);
+  });
+
+  afterEach(() => {
+    MockDate.reset();
+  });
+
+  /**
+   * Simplest case, empty data
+   */
+  it('intersect of empty locations produces bins with no intersections', () => {
```

Comment:
Suggest a little more testing on the possible edge cases around the 0 daybin (today).. like end of day, start of day, middle of day.

Reply:
Yeah, a test that crosses midnight does make sense.  I'll add.